kolibri/plugins/setup_wizard/assets/src/api.js

```diff
@@ -58,3 +60,16 @@ export const SetupSoUDTasksResource = new Resource({
     });
   },
 });
+
+export const FinishSoUDSyncingResource = new Resource({
+  name: 'restartzeroconf',
+  namespace: 'kolibri.plugins.setup_wizard',
+  finish() {
+    const welcomeDimissalKey = 'DEVICE_WELCOME_MODAL_DISMISSED';
+    const device_url = urls['kolibri:kolibri.plugins.device:device_management'];
+    window.sessionStorage.setItem(welcomeDimissalKey, false);
+    this.postListEndpoint('restart');
+    redirectBrowser(device_url());
```

Comment:
In case `device_url` is `undefined`, we should change this to:
```suggestion
    redirectBrowser(device_url ? device_url(): null);
```

Reply:
@rtibbles  maybe this is a naive question, but how could `device_url` be null? are there any case where we are not loading the `device` plugin?

Reply:
If a user has disabled the device plugin then this would be null. In general, I have preferred to enforce any coupling between our plugins to be loose, so that one does not cause errors because another is disabled.

Reply:
ok, I never thought this would be an use case but, as usual, you're right.

But... another naive question, shouldn't we add this guard to https://github.com/learningequality/kolibri/blob/release-v0.15.x/kolibri/core/assets/src/core-app/urls.js#L14 ? or making it fail is being used in the code?

Reply:
I would like to add a guard that just returned a function for every URL, so that we could always invoke the function, and just return defined in that case, but the best way to achieve that would be with an [ES6 Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) object, which is not available on IE11 and not polyfillable.

The function you are pointing to there is only used by the URL functions defined in our Javascript, rather than the JS generated by Django-js-reverse (we could in theory hack that template to do this, but it felt like a lot of work!).